prunner.go

```diff
@@ -392,7 +394,11 @@ func (r *PipelineRunner) HandleStageChange(stage *scheduler.Stage) {
 		return
 	}
 
-	jt.Status = toStatus(stage.ReadStatus())
+	if jt.Canceled {
```

Comment:
Here the task flag is "promoted" to the task status which is set by stage updates from the scheduler.